AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m

```diff
@@ -96,31 +120,67 @@ - (void)channel:(id<MSChannelProtocol>)channel didSetEnabled:(BOOL)isEnabled and
-  if (![self shouldSendLogToOneCollector:log] ||
-      ![channel conformsToProtocol:@protocol(MSChannelUnitProtocol)]) {
-    return;
-  }
-  id<MSChannelUnitProtocol> channelUnit = (id<MSChannelUnitProtocol>)channel;
-  NSString *groupId = channelUnit.configuration.groupId;
-  id<MSChannelUnitProtocol> oneCollectorChannelUnit = [self.oneCollectorChannels objectForKey:groupId];
-  if (!oneCollectorChannelUnit) {
-    return;
-  }
-  id<MSLogConversion> logConversion = (id<MSLogConversion>)log;
-  NSArray<MSCommonSchemaLog *> *commonSchemaLogs = [logConversion toCommonSchemaLogs];
-  for (MSCommonSchemaLog *commonSchemaLog in commonSchemaLogs) {
-    [oneCollectorChannelUnit enqueueItem:commonSchemaLog];
-  }
-}
+#pragma mark - Helper
 
 - (BOOL)isOneCollectorGroup:(NSString *)groupId {
   return [groupId hasSuffix:kMSOneCollectorGroupIdSuffix];
 }
 
 - (BOOL)shouldSendLogToOneCollector:(id<MSLog>)log {
   NSObject *logObject = (NSObject *)log;
-  return [[log transmissionTargetTokens] count] > 0 && [log conformsToProtocol:@protocol(MSLogConversion)] && ![logObject isKindOfClass:[MSCommonSchemaLog class]];
+  return [[log transmissionTargetTokens] count] > 0 && [log conformsToProtocol:@protocol(MSLogConversion)] &&
+         ![logObject isKindOfClass:[MSCommonSchemaLog class]];
+}
+
+- (BOOL)validateLog:(MSCommonSchemaLog *)log {
+  if (![self validateLogName:log.name]) {
+    return NO;
+  }
+  if (![self validateLogData:log.data]) {
+    return NO;
+  }
+  return YES;
+}
+
+- (BOOL)validateLogName:(NSString *)name {
+
+  // Name mustn't be nil.
+  if (!name.length) {
+    MSLogError([MSAppCenter logTag], @"%@ Name must not be nil or empty.", kMSBaseErrorMsg);
+    return NO;
+  }
+
+  // The Common Schema event name must conform to a regex.
+  NSError *error = nil;
+  NSRegularExpression *regex =
+      [NSRegularExpression regularExpressionWithPattern:kMSLogNameRegex options:0 error:&error];
+  NSRange range = NSMakeRange(0, name.length);
+  if (!regex) {
+    MSLogError([MSAppCenter logTag], @"%@ Couldn't create regular expression with pattern \"%@\": %@", kMSBaseErrorMsg,
+               kMSLogNameRegex, error.localizedDescription);
+    return NO;
+  }
+  NSUInteger count = [regex numberOfMatchesInString:name options:0 range:range];
+  if (!count) {
+    MSLogError([MSAppCenter logTag], @"%@ Name must match '%@' but was '%@'", kMSBaseErrorMsg, kMSLogNameRegex, name);
+    return NO;
+  }
+  return YES;
+}
+
+- (BOOL)validateLogData:(MSCSData *)data {
+  NSDictionary<NSString *, NSString *> *properties = data.properties;
+  for (NSString *key in properties) {
+    if (![key isKindOfClass:[NSString class]] || ![properties[key] isKindOfClass:[NSString class]]) {
+      MSLogError([MSAppCenter logTag], @"%@ Properties key and value must be of NSString.", kMSBaseErrorMsg);
+      return NO;
+    }
+    if (!key.length || !properties[key].length) {
```

Comment:
We allow empty strings in Android since it's allowed in both One Collector and Aria (and it's visible in live sampled events, including the mapping empty key to empty value is displayed with a blank line...). Only null should be checked.

Reply:
I simply removed the check then, we just can't have null objects in a NSDictionary.